mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -106,6 +142,29 @@ pub struct S3PutObjectRequest {
     review_callback: ReviewCallbackBox,
     start_time: Instant,
     total_bytes: u64,
+    response_headers: Arc<Mutex<Option<Headers>>>,
+    pub expected_headers: Vec<(String, String)>,
```

Comment:
Also, why is it `pub`?

Reply:
Well, I anticipated this question, it is [accessed in a test](https://github.com/awslabs/mountpoint-s3/pull/693/files#diff-8686057abaa298b6e3d57ea579a193b1b37a934ab45fb5434159abfc0684bf89R426) for `check_headers` function. I honestly don't know a better way of testing this check (given S3 never returns unexpected headers) than altering the internal state of the `S3PutObjectRequest`. I'm open for suggestions here.

Reply:
The other way around is to write a unit test for `check_headers` in the same file (and get rid of current tests for this). It won't be checking the actual CRT behaviour though (e.g. CRT may start filtering out those headers with some new release, idk).

Reply:
It is kind of a red flag for an integration test to need altering the internal state of an object. I think these fields should not be public and the test should move to this module where:
* it can test only `check_headers`, as you mention
* or try and build your artificial `S3PutObjectRequest` instance and call `complete()` on it